src/pages/index.js

```diff
@@ -3,32 +3,99 @@ import { FormValidator } from "../componets/FormValidator.js";
 import Section from "../componets/Section.js";
 import PopupWithImage from "../componets/PopupWithImage.js";
 import PopupWithForm from "../componets/PopupWithForm.js";
+import PopupWithConfirmation from "../componets/PopupWithConfirmtaion.js";
 import UserInfo from "../componets/UserInfo.js";
-import { initialCards, formSettings } from "../utils/constants.js";
+import { formSettings, token } from "../utils/constants.js";
 import "./index.css";
+import Api from "../componets/Api.js";
+
+// Profile consts
+const profile = document.querySelector(".profile");
+const profileName = profile.querySelector(".profile__name");
+const profileDecsription = profile.querySelector(".profile__description");
+const profileAvatar = profile.querySelector(".profile__avatar");
+
+const profileInfo = new UserInfo({
+  name: profileName,
+  description: profileDecsription,
+  avatar: profileAvatar,
+});
+
+// Cards and Cardlist
+const cardListLocation = document.querySelector(".cards");
+
+function deleteCard(cardId, card) {
+  return api
+    .deleteCard(cardId)
+    .then((res) => {
+      if (res.ok) {
+        card.remove();
+        return Promise.resolve();
+      } else {
+        return Promise.reject(res.status);
+      }
+    })
+    .catch((err) => {
+      console.error(`ERROR: Failed to delete card: ${err}`);
+    });
+}
+
+function handleCardLike(isLiked, cardId) {
+  if (isLiked) {
+    return api.removeCardLike(cardId).then((res) => {
+      if (res.ok) {
+        return Promise.resolve();
+      } else {
+        return Promise.reject(`Failed to remove like: ${res.status}`);
+      }
+    });
+  } else {
+    return api.addCardLike(cardId).then((res) => {
+      if (res.ok) {
+        return Promise.resolve();
+      } else {
+        return Promise.reject(`Failed to add like: ${res.status}`);
+      }
+    });
+  }
+}
+
+function confirmCardDelete(card) {
+  card.deleteCard().finally(() => {
+    cardDeleteModal.close();
```

Comment:
You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well